types/index.d.ts

```diff
@@ -0,0 +1,224 @@
+export interface StartOptions {
+    /**
+     * Should annyang restart itself if it is closed indirectly, because of silence or window conflicts?
+     *
+     * @type {boolean}
+     */
+    autoRestart?: boolean;
+    /**
+     * Allow forcing continuous mode on or off. Annyang is pretty smart about this, so only set this if you know what you're doing.
+     *
+     * @type {boolean}
+     */
+    continuous?: boolean;
+}
+
+/**
+ * A command option that supports custom regular expressions
+ *
+ * @export
+ * @interface CommandOptionRegex
+ */
+export interface CommandOptionRegex {
+    /**
+     * @type {RegExp}
+     */
+    regexp: RegExp;
+    /**
+     * @type {() => any}
+     */
+    callback(): void;
+}
+
+/**
+ * Commands that annyang should listen to
+ *
+ * #### Examples:
+ * ````javascript
+ * {'hello :name': helloFunction, 'howdy': helloFunction};
+ * {'hi': helloFunction};
+ * ````
+ * @export
+ * @interface CommandOption
+ */
+export interface CommandOption {
+    [command: string]: CommandOptionRegex | (() => void);
+}
+
+/**
+ * Supported Events that will be triggered to listeners, you attach using `annyang.addCallback()`
+ *
+ * `start` - Fired as soon as the browser's Speech Recognition engine starts listening
+ * `error` - Fired when the browser's Speech Recogntion engine returns an error, this generic error callback will be followed by more accurate error callbacks (both will fire if both are defined)
+ * `errorNetwork` - Fired when Speech Recognition fails because of a network error
+ * `errorPermissionBlocked` - Fired when the browser blocks the permission request to use Speech Recognition.
+ * `errorPermissionDenied` - Fired when the user blocks the permission request to use Speech Recognition.
+ * `end` - Fired when the browser's Speech Recognition engine stops
+ * `result` - Fired as soon as some speech was identified. This generic callback will be followed by either the `resultMatch` or `resultNoMatch` callbacks.
+ *     Callback functions registered to this event will include an array of possible phrases the user said as the first argument
+ * `resultMatch` - Fired when annyang was able to match between what the user said and a registered command
+ *     Callback functions registered to this event will include three arguments in the following order:
+ *       * The phrase the user said that matched a command
+ *       * The command that was matched
+ *       * An array of possible alternative phrases the user might've said
+ * `resultNoMatch` - Fired when what the user said didn't match any of the registered commands.
+ *     Callback functions registered to this event will include an array of possible phrases the user might've said as the first argument
+ */
+export type Events =
+    'start' |
+    'soundstart' |
+    'error' |
+    'end' |
+    'result' |
+    'resultMatch' |
+    'resultNoMatch' |
+    'errorNetwork' |
+    'errorPermissionBlocked' |
+    'errorPermissionDenied';
+
+export interface Annyang {
```

Comment:
Is export interface the correct way to do this?
Looking at this [example](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html), and at the footnote [Preventing Name Conflicts](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html), it looks like we might actually want to define a namespace annyang.

Am I understanding this wrong?